src/overlay/FlowControl.h

```diff
@@ -146,60 +159,67 @@ class FlowControl
     size_t
     getTxQueueByteCountForTesting() const
     {
+        MutexLocker lockGuard(mFlowControlMutex);
```

Comment:
Nit: These getters and setters should be annotated by `LOCKS_EXCLUDED`. Just currious, were these locks added based on the static analysis feedback?

Reply:
I was debating about annotating test-only code. The way we write tests, these functions are always executed from main thread only, when nothing else is running, so these annotations are kind of useless (we're forced into this partially because Catch is not thread-safe). But without annotations we get warnings, which creates confusion. I can add LOCKS_EXCLUDED for completeness, but just wanted to flag that for test-only code annotations are more of a code bloat rather than something meaningful.

Reply:
True, but for something important like this, I think there's good value in having no warning messages vs. warning messages that we always ignore because their in tests. Once we get in the habit of ignoring warnings, I imagine a non-test warning might slip through